Fix compilation on Windows

mohits opened the pull request:
Update fast_polylines.c to include definition of uint so that it doesn't fail when building on Ruby2.6 on Windows (at least).

Commit:
Update fast_polylines.c to include definition of uint so that it doesn't fail when building on Ruby2.6 on Windows (at least).

mohits committed:
Update version.rb

BuonOmo commented:
Hi @mohits,

Thanks for the contribution, it looks great and mandatory!

However there are 2 issues for me to be able to merge it: 1. please do not bump the version, we'll do it when publishing!

And I guess there is another issue, we cannot ensure that regression is avoided on this since our CI is only running on Linux and MacOS. A possible option is to add windows to the CI:

```yml
language: ruby
os:
  - linux
  - osx
  - windows
rvm:
  - 2.4
  - 2.5
  - 2.6
  - 2.7
script: make test
cache: bundler
```

Could you add this to your PR so we'll have a check for windows as well ?

Cheers,
Ulysse

P.S: Feel free to assign or mention me when you're done or you have a question :)

mohits commented:
Hi @BuonOmo - sorry for that. It was originally meant to have the new version only on my side so that I could distinguish it from your release. I will get back to this in a couple of days and re-submit the pull request.

(happy to hear that you will eventually merge it in cos then I don't have to point my Gemfile to my "fake" version)

Commit:
Revert "B fix compile windows (#1)"

This reverts commit 115e5d577315f503a14f75a36bdb0f669404e7cb.

Commit:
Merge branch 'b_fix_compile_windows' of https://github.com/mohits/fast-polylines.git

Commit:
Resolve PR comments
- restore version back to 2.1.0 (so that it can be changed in the final release)
- added the unsigned int typedef (so that it compiles on Windows)
- add windows to the Travis CI file

mohits commented:
Seems we have a problem getting the CI to run Windows :(
> The language 'ruby' is currently unsupported on the Windows Build Environment.

Let me know if you want me to kill that.

BuonOmo commented:
@mohits looking a bit further it seems that it's been going on a while and nothing has been done on the side of Travis CI (https://travis-ci.community/t/ruby-support-on-windows/265).

I guess we could remove this test, or make it failure-allowed. And add a comment on top of the typedef explaining that it is required for this to work on windows. And I think I should specify in the README that windows support is not guaranteed?

Does it seems ok for you ? If it's the case I'll integrate your PR next week. Otherwise, we'll have to wait for a windows ready CI (which I think won't happen before long).

mohits commented:
@BuonOmo - I agree.

We should remove the Windows check in the CI and update the Readme to say:
-> fast_polylines should compile and work on Windows for Ruby >= 2.4 but is not verified on CI due to lack of support for Ruby on Windows. If you find a problem, feel free to open an issue here so that we can look at it.

If someone opens an issue, just tag me to it since I do have a Windows machine here where I am using it to take a closer look.

Let's get this merged. 

Once done, I have another:
- minor fix which is preventing me from running the benchmark on my computer
- minor enhancement which speeds up the benchmark by 5% ~ 20% (depending on different runs) by changing the C code just a little bit ;-)

Cheers!

BuonOmo commented:
@mohits I'm all for that!

Could you:
- add a comment above the typedef mentioning this is for windows
- remove windows from travis

And then I'll merge that one :)

mohits committed:
Completing the changes required for Windows compilation
- Added a comment to indicate that the uint typedef is needed on Windows
- Updated .travis.yml to remove Windows since Travis does not yet properly suppory Windows for Ruby
